app/data_providers/snowplow_corpus_slate_tracker.py

```diff
@@ -0,0 +1,103 @@
+import os
+from abc import ABC
+from typing import List, Any, Dict, Optional
+
+from aws_xray_sdk.core import xray_recorder
+from snowplow_tracker import Tracker, Subject, SelfDescribingJson, Emitter
+
+from app.models.corpus_slate_model import CorpusSlateModel
+from app.config import ENV, ENV_PROD
+
+
+class CorpusSlateTrackable(ABC):
+    """
+    Abstract class to track the recommendation of a CorpusSlate.
+    """
+
+    async def track(self, corpus_slate: CorpusSlateModel, user_id: Optional[str]):
+        return NotImplemented
+
+
+class SnowplowConfig:
+    APP_ID = f'pocket-data-products-recommendation-api-{ENV}'
+
+    PROD_ENDPOINT_URL = 'com-getpocket-prod1.collector.snplow.net'
+    DEV_ENDPOINT_URL = 'com-getpocket-prod1.mini.snplow.net'
+    ENDPOINT_URL = PROD_ENDPOINT_URL if ENV == ENV_PROD else DEV_ENDPOINT_URL
+
+    CORPUS_SLATE_SCHEMA = 'iglu:com.pocket/corpus_slate/jsonschema/1-0-0'
+    USER_SCHEMA = 'iglu:com.pocket/user/jsonschema/1-0-0'
+    OBJECT_UPDATE_SCHEMA = 'iglu:com.pocket/object_update/jsonschema/1-0-7'
+
+
+class SnowplowCorpusSlateTracker(CorpusSlateTrackable):
+    """
+    Implements tracking the recommendation of a CorpusSlate in SnowPlow, with the goal of tracking any metadata that
+    to CorpusRecommendation.id that we are interested in. Clients will only emit the CorpusRecommendation id in events.
+    """
+
+    def __init__(self, tracker: Tracker, snowplow_config: SnowplowConfig):
+        self.tracker = tracker
+        self.snowplow_config = snowplow_config
+
+    @xray_recorder.capture_async('data_providers.SnowplowCorpusSlateTracker.track')
+    async def track(self, corpus_slate: CorpusSlateModel, user_id: Optional[str]):
```

Comment:
why user_id is optional? is there a use case when we don't have it? is it also optional in the schema?